File: php/src/Types/ClientConfig.php
<?php

declare(strict_types=1);

namespace Royaltics\ErrorTracker\Types;

final readonly class ClientConfig
{
    public function __construct(
        public string $webhookUrl,
        public string $licenseId,
        public string $licenseDevice,
        public ?string $licenseName = null,
        public ?string $app = null,
        public ?string $version = null,
        public ?string $platform = null,
        public bool $enabled = true,
        public int $maxRetries = 3,
        public int $timeout = 10000,
        public int $flushInterval = 5000,
        public int $maxQueueSize = 50,
        public array $headers = []
    ) {
        $this->validate();
    }

    private function validate(): void
    {
        if (empty($this->webhookUrl)) {
            throw new \InvalidArgumentException('webhookUrl is required');
        }

        if (!filter_var($this->webhookUrl, FILTER_VALIDATE_URL)) {
            throw new \InvalidArgumentException('webhookUrl must be a valid URL');
        }

        if (empty($this->licenseId)) {
            throw new \InvalidArgumentException('licenseId is required');
        }

        if (empty($this->licenseDevice)) {
            throw new \InvalidArgumentException('licenseDevice is required');
        }

        if ($this->maxRetries < 0 || $this->maxRetries > 10) {
            throw new \InvalidArgumentException('maxRetries must be between 0 and 10');
        }

        if ($this->timeout < 1000 || $this->timeout > 60000) {
            throw new \InvalidArgumentException('timeout must be between 1000 and 60000ms');
        }

        if ($this->flushInterval < 100) {
            throw new \InvalidArgumentException('flushInterval must be at least 100ms');
        }

        if ($this->maxQueueSize < 1) {
            throw new \InvalidArgumentException('maxQueueSize must be at least 1');
        }
    }
}

File: csharp/Types/ClientConfig.cs
namespace Royaltics.ErrorTracker.Types;

public sealed record ClientConfig
{
    public required string WebhookUrl { get; init; }
    public required string LicenseId { get; init; }
    public required string LicenseDevice { get; init; }
    public string? LicenseName { get; init; }
    public string? App { get; init; }
    public string? Version { get; init; }
    public string? Platform { get; init; }
    public bool Enabled { get; init; } = true;
    public int MaxRetries { get; init; } = 3;
    public int Timeout { get; init; } = 10000;
    public int FlushInterval { get; init; } = 5000;
    public int MaxQueueSize { get; init; } = 50;
    public Dictionary<string, string> Headers { get; init; } = new();

    public void Validate()
    {
        if (string.IsNullOrWhiteSpace(WebhookUrl))
            throw new ArgumentException("WebhookUrl is required", nameof(WebhookUrl));

        if (!Uri.TryCreate(WebhookUrl, UriKind.Absolute, out var uri) || 
            (uri.Scheme != Uri.UriSchemeHttp && uri.Scheme != Uri.UriSchemeHttps))
            throw new ArgumentException("WebhookUrl must be a valid HTTP/HTTPS URL", nameof(WebhookUrl));

        if (string.IsNullOrWhiteSpace(LicenseId))
            throw new ArgumentException("LicenseId is required", nameof(LicenseId));

        if (string.IsNullOrWhiteSpace(LicenseDevice))
            throw new ArgumentException("LicenseDevice is required", nameof(LicenseDevice));

        if (MaxRetries < 0 || MaxRetries > 10)
            throw new ArgumentException("MaxRetries must be between 0 and 10", nameof(MaxRetries));

        if (Timeout < 1000 || Timeout > 60000)
            throw new ArgumentException("Timeout must be between 1000 and 60000ms", nameof(Timeout));

        if (FlushInterval < 100)
            throw new ArgumentException("FlushInterval must be at least 100ms", nameof(FlushInterval));

        if (MaxQueueSize < 1)
            throw new ArgumentException("MaxQueueSize must be at least 1", nameof(MaxQueueSize));
    }
}
